Sources/Metrics/Metrics.swift

```diff
@@ -74,3 +74,39 @@ extension Timer {
         }
     }
 }
+
+#if swift(>=5.7)
+
+public enum TimerError: Error {
+    case durationToIntOverflow
+}
+
+extension Timer {
+    /// Convenience for recording a duration based on ``Duration``.
+    ///
+    /// `Duration` will be converted to an `Int64` number of nanoseconds, and then recorded with nanosecond precision.
+    ///
+    /// - Parameters:
+    ///     - duration: The `Duration` to record.
+    ///
+    /// - Throws: `TimerError.durationToIntOverflow` if conversion from `Duration` to `Int64` of Nanoseconds overflowed.
+    @available(macOS 13, iOS 16, tvOS 15, watchOS 8, *)
+    @inlinable
+    public func record(_ duration: Duration) throws {
```

Comment:
I am not convinced we want these to throw as it will make the call sites awkward to use. I think we should record Int.max instead as we do in `recordNanoseconds`

Reply:
exactly. we already take that approach in `recordXXX`, eg

```

    @inlinable
    public func recordNanoseconds<DataType: BinaryInteger>(_ duration: DataType) {
        self.recordNanoseconds(duration >= Int64.max ? Int64.max : Int64(duration))
    }

    @inlinable
    public func recordSeconds<DataType: BinaryInteger>(_ duration: DataType) {
        guard duration <= Int64.max else { return self.recordNanoseconds(Int64.max) }

        let result = Int64(duration).multipliedReportingOverflow(by: 1_000_000_000)
        if result.overflow {
            self.recordNanoseconds(Int64.max)
        } else {
            self.recordNanoseconds(result.partialValue)
        }
    }
```

Reply:
@tomerd, pushed up a change.

Performing overflow checks one at a time to not perform the second operation if the first already overflowed.

I don't yet know the separation into `CoreMetrics` and plain `Metrics` directories. How do you split them? My gut feeling was that it's about cross-platform support, band `CoreMetrics` is designed to work everywhere where, only relies on standard library, but `Metrics` uses `Foundation` as well? 

Is it important where that Timer extension lives?

Reply:
Aside: I know an engineer at Apple could put that PR and merge it in under an hour, and I appreciate your patience and guidance, @tomerd.